create_lmodrc.py

```diff
@@ -17,6 +17,113 @@
 }
 """
 
+GPU_LMOD_RC ="""require("strict")
+local hook = require("Hook")
+local open = io.open
+
+local function read_file(path)
+    local file = open(path, "rb") -- r read mode and b binary mode
+    if not file then return nil end
+    local content = file:read "*a" -- *a or *all reads the whole file
+    file:close()
+    return content
+end
+
+-- from https://stackoverflow.com/a/40195356
+--- Check if a file or directory exists in this path
+function exists(file)
```

Comment:
This function didn't work for checking the `libcuda.so` in `/.singularity.d`. I modified it to:
```
function exists(file)
    local ok, err, code = os.rename(file, file)
    if not ok then
        io.stderr:write(string.format("Error opening file %s. Error msg: %s. Error code: %d\n", file, err, code))
        if code == 13 then
            -- Permission denied, but it exists
            return true
        end
    end
    return ok, err
end
```
to figure out what was going on and got:
```
Error opening file /.singularity.d/libs/libcuda.so. Error msg: /.singularity.d/libs/libcuda.so: Read-only file system. Error code: 30
```
So, there are more exit codes that should _probably_ pass. 

The most robust solution might be better to find another way to check if a file exists, but I couldn't find any that doesnt either 1. try to open the file in read mode, 2. tries to rename (as you do) or 3. requires a non-standard LUA packages (`lsf`  which can use `lsf.attributes` apparently to check for file existence, or `luaposix` which can do `stat` calls).

Since all of those are annoying, maybe the best thing to do for now is just add this error code. At least it'll tackle the specific case of this being a read-only dir.

Reply:
Fixed by using native Lmod capabilities